### PR TITLE
ENH: use crs.is_projected and heuristic as fallback (#1713)

### LINKED ISSUE #1713: REF: remove heuristics in __repr__ guessing if CRS if projected
In `GeometryArray.__repr__` we have a heuristic which tries to guess if a geometry is projected or not based on coords of first ten geometries. The reason is that we set rounding of geographic CRS to 5 while for projected to 3.

https://github.com/geopandas/geopandas/blob/59d3b2b7260b485e95e06a38f1c3e218b0a40a46/geopandas/array.py#L1005-L1023

We can remove this and use `crs.is_projected` or `crs.is_geographic` instead. Something along

```py
if precision is None:
    if self.crs and self.crs.is_projected:
        precision = 3
    else:
        precision = 5
```

### PR DESCRIPTION
**Pull request by mesejo**
Closes #1713

### CODE REVIEW

File: geopandas/array.py [hunk 1]
```diff
@@ -1224,23 +1224,27 @@ def _formatter(self, boxed=False):
...
-                    and (-90 <= ymin <= 90)
-                    and (-90 <= ymax <= 90)
-                ):
-                    # geographic coordinates
-                    precision = 5
-                else:
-                    # typically projected coordinates
-                    # (in case of unit meter: mm precision)
+                if self.crs and self.crs.is_projected:
                     precision = 3
```
Comment (martinfleis) [inline geopandas/array.py:1228]:
```suggestion
                if self.crs
                    if self.crs.is_projected:
                        precision = 3
                    else:
                        precision = 5
```

No need to go through fallback for a known geographic projection.

**Review by martinfleis — COMMENTED** [review]
Thanks a lot! 

Just a minor suggestion above.

File: geopandas/array.py [hunk 2]
```diff
@@ -1224,23 +1224,30 @@ def _formatter(self, boxed=False):
...
-                    and (-90 <= ymin <= 90)
-                    and (-90 <= ymax <= 90)
-                ):
-                    # geographic coordinates
-                    precision = 5
+                if self.crs:
+                    if self.crs.is_projected:
+                        precision = 3
+                    else:
+                        precision = 5
```
Comment (martinfleis) [inline geopandas/array.py:1231]: It seems that we don't have this option tested currently. Can you add one simple test for `__repr__` based on geometry in geographic projection (4326)?

Comment (martinfleis) [reply geopandas/array.py:1231]: Here https://github.com/geopandas/geopandas/blob/2a5c42f8fcf7093caf5e59c012ed05817b68569f/geopandas/tests/test_pandas_methods.py#L42

Comment (mesejo) [reply geopandas/array.py:1231]: Sure thing! What does the 4326 means?

Comment (martinfleis) [reply geopandas/array.py:1231]: Actually, it would be good to explicitly check both options based on CRS since the existing tests all fall to heuristic.

Comment (mesejo) [reply geopandas/array.py:1231]: Should I add a test for the case when the crs is projected?

Comment (martinfleis) [reply geopandas/array.py:1231]: Yes. Two tests are needed. One with some explicit projected CRS and one with explicit geographic CRS.

**Review by martinfleis — APPROVED** [review]
Thanks @mesejo! Looks perfect!

**Pull request merged**